### DISCUSSION TITLE
How do we encapsulate the class id information with bbox?

### DISCUSSION THREAD

**Discussion by shreyas-kamath-simplisafe** [kornia/kornia#2364] [Potential Issue]
Hi, 

I have been trying to incorporate Kornia for object detection purposes. Since intensity-based transformations do not require bboxes, those work well. However, when I try to use geometry-based transforms, how do correlate between bbox and its corresponding class id in `AugmentationSequential`? I see a `CLASS` in `DataKey` however, I keep getting the following error
```
Exception has occurred: NotImplementedError       (note: full exception trace is shown but execution is paused at: _run_module_as_main)
input type of DataKey.CLASS is not implemented.
```

Can someone help me with this?

**Comment by edgarriba** [comment]
@shreyas-kamath-simplisafe could you share how do you use the api ? with a small snippet

**Reply by shreyas-kamath-simplisafe** [reply]
@edgarriba  This is a basic example that I took from one of the tutorials:

```
import matplotlib.pyplot as plt
import kornia.augmentation as K
import torch
import io
import cv2
import requests
from kornia.utils import image_to_tensor, tensor_to_image

import numpy as np

# https://kornia-tutorials.readthedocs.io/en/latest/_nbs/data_augmentation_mosiac.html#install-and-get-data
def download_image(url: str, filename: str = "") -> str:
    filename = url.split("/")[-1] if len(filename) == 0 else filename
    # Download
    bytesio = io.BytesIO(requests.get(url).content)
    # Save file
    with open(filename, "wb") as outfile:
        outfile.write(bytesio.getbuffer())

    return filename


url = "https://raw.githubusercontent.com/kornia/data/main/panda.jpg"
download_image(url)

bbox = [torch.tensor([[355, 10, 660, 250]])]
img = cv2.imread("panda.jpg", cv2.IMREAD_COLOR)
img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
h, w = img.shape[:2]
img_tensor = image_to_tensor(img).float() / 255.0

inputs = [img_tensor, bbox]

aug = K.AugmentationSequential(
    K.Resize((300, 500)), data_keys=['input', 'bbox_xyxy'])

transformed = aug(*inputs)`
```


If you look at this example, the `bbox` is a `1x4` array without any class labels. Since it is Resize I do not have any problem, But if I do `CenterCrop` or any other geometric transformation, there might be a chance that the bbox with certain labels be out of bounds. In such scenarios how do we figure it out which bbox relates to which class in a nulticlass object detection model??

**Reply by edgarriba** [reply]
the only way i see to detect such cases is via projecting / reprojecting. You can access to the transform with `aug.transform_matrix` after every call. Then you can use `kornia.geometry.transform_points` to do your own verification logic.

**Reply by shijianjian** [reply]
Another option is to split it into `data_keys=['input', 'bbox_xyxy', 'class']` then merge it together.

I think this class-box is a feature that worth adding but not sure about the proper path to go.
